src/Sentry/Platforms/Native/SentryNative.cs

```diff
@@ -17,7 +17,7 @@ internal static class SentryNative
     // This way, `SentryNative.IsEnabled` should be treated as a compile-time constant for trimmed apps.
     [FeatureSwitchDefinition(SentryNativeIsEnabledSwitchName)]
 #endif
-    private static bool IsEnabled => !AppContext.TryGetSwitch(SentryNativeIsEnabledSwitchName, out var isEnabled) || isEnabled;
+    private static bool IsEnabled => AppContext.TryGetSwitch(SentryNativeIsEnabledSwitchName, out var isEnabled) && isEnabled;
```

Comment:
praise:
If I understand it right, we're changing the behavior - in conjunction with the changes in `Sentry.Native.targets`
- from `true` when `Sentry.Native.IsEnabled` is _not_ set **or** explicitly set to `true`
- to `true` when `Sentry.Native.IsEnabled` _is_ set **and** set to `true`

I do find that more clear than before ... it's more of a `false` by default, unless we detect it's supported, unless the consuming projects opts out via `SentryNative`.